### PR TITLE
Create trips

### PR DESCRIPTION
**Pull request by juliemahlke**
Create a form-page to add a new trip for travel-wishes or/and planned trips

### CODE REVIEW

**Comment by arisslin** [comment]
After cloning the project i started 'npm install' and get a lot of errors and at the end i got the log below:

```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@fortawesome%2fpro-solid-svg-icons - Not found
npm ERR! 404
npm ERR! 404  '@fortawesome/pro-solid-svg-icons@5.12.1' is not in the npm registry.
npm ERR! 404 You should bug the author to publish it (or use the name yourself!)
npm ERR! 404 It was specified as a dependency of 'capstone-project'
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/arisslin/.npm/_logs/2020-03-07T15_27_41_517Z-debug.log
```

The node_modules folder is empt so it stops to get all dependencies. Looks like there is something went wrong with font-awesome

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
-import styled from 'styled-components'
+import styled from 'styled-components/macro'
+import CreateTrip from './components/CreateTrip'
+import TripsData from './data/tripsdata.json'
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
+import Icons from './common/Icons'
+import ButtonAdd from './components/ButtonAdd'
 
 function App() {
+  const [trips, setTrips] = useState(TripsData)
```
Comment (arisslin) [inline src/App.js]: Maybee you can check before if there are trip datas and if not asign an empty array:
```
const tripsData = TripsData ? TripsData : []
const [trips, setTrips] = useState(TripsData)
````
Only an idea, no must.

File: src/components/ButtonAdd.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react'
+import { BrowserRouter as Router, Link } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import styled from 'styled-components'
+import { NavLink } from 'react-router-dom'
+
+export default function ButtonAdd() {
+  const [toggle, setToggle] = useState(true)
```
Comment (arisslin) [inline src/components/ButtonAdd.js]: Maybee it is better to place the hook into the app.js?
I am not sure what is best practice.

File: src/components/ButtonAdd.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  useEffect(() => {}, [toggle])
```
Comment (arisslin) [inline src/components/ButtonAdd.js]: There is no further code so useEffect dos nothing. You can delete this line.

File: src/components/TripListItem.js [hunk 4]
```diff
@@ -2,14 +2,14 @@ import React from 'react'
 import styled from 'styled-components/macro'
 import defaultImg from '../images/default-image.jpg'
 
-export default function TripListItem({ title, date }) {
```
Comment (arisslin) [inline src/components/TripListItem.js]: I think when you only have 2 props than you do not need an object.

**Comment by juliemahlke** [comment]
Oh ok... I think you need to install font awesome:
npm i --save @fortawesome/fontawesome-svg-core
npm i --save @fortawesome/react-fontawesome
npm i --save @fortawesome/pro-solid-svg-icons 
npm i --save @fortawesome/pro-regular-svg-icons
npm i --save @fortawesome/pro-light-svg-icons

**Pull request merged**